src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -174,6 +174,7 @@ public class ImageSharpMiddleware
                 using (var image = FormattedImage.Load(this.options.Configuration, inBuffer))
                 {
                     image.Process(this.logger, this.processors, commands);
+                    this.options.OnBeforeSave(image);
```

Comment:
probably want to call this using this pattern `this.options.OnBeforeSave?.Invoke(image);` in the event `OnBeforeSave` is ever manually set to null.

Reply:
We probably want to make sure we do that for all the extension points form the `options`

Reply:
I don't like the idea of swallowing up exceptions like that. If someone manually sets it to null it should throw. 

I'm actually following the pattern from the MS staticfile middleware.

Reply:
Its not an exception really as you wouldn't be swallowing anything its just skipping running the delegate if its but unset.

Reply:
Here's what I mean. 

https://github.com/aspnet/StaticFiles/blob/826e89a2d1b325485819040ff41b9948afc91e06/src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs#L285

Can't find a unit test to copy either. We do need to do some others though.